addons/mail/static/src/new/core/messaging_service.js

```diff
@@ -540,6 +541,7 @@ export class Messaging {
                             ...notif.payload.persona.partner,
                             ...notif.payload.persona.guest,
                             type: notif.payload.persona.partner ? "partner" : "guest",
+                            threadId: notif.payload.persona.guest ? channel.id : null,
```

Comment:
I just renamed `threadId` to `channelId`, to be more specific (it's a bit more clear the intend of this field with this rename)